[Chiam Yun Qing] iP

chiamyunqing opened the pull request.

Commit:
Add Gradle support

chiamyunqing committed:
Add the greeting and exit codes

chiamyunqing committed:
Allow Duke to echo input from users

chiamyunqing committed:
Remove greet and exit function calls in echo function to make it independent

chiamyunqing committed:
Instantiate Duke object in main() instead

chiamyunqing committed:
Able to record tasks (i.e. add and list tasks)

chiamyunqing committed:
Able to record tasks (i.e. add and list tasks)

chiamyunqing committed:
Split the processing of tasks into parts, enable mark done task ability

chiamyunqing committed:
Add Todo, Deadline and Event classes

chiamyunqing committed:
Update ToDo, Event and Deadline classes; Add feature to process different tasks

chiamyunqing committed:
Update printing of various tasks by changing toString() method

chiamyunqing committed:
Test for different commands added

chiamyunqing committed:
Convert Task class to abstract - rename tag, passed test cases

chiamyunqing committed:
Add all the Duke exception classes

chiamyunqing committed:
Able to handle errors (i.e. invalid task number, invalid task, empty description)

chiamyunqing committed:
Add feature to delete tasks

chiamyunqing committed:
Update string format of task classes to make handling of loading tasks easier, Add storage class for handling saving and loading of data

chiamyunqing committed:
Add functionality for saving task and loading task from hard disk

chiamyunqing committed:
Create new data folder to store duke.txt

chiamyunqing committed:
Add feature to process date and time for events and deadlines

chiamyunqing committed:
Add command to enable printing of tasks occurring on a specific date

chiamyunqing committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Create new data folder to store duke.txt
  Add functionality for saving task and loading task from hard disk
  Update string format of task classes to make handling of loading tasks easier, Add storage class for handling saving and loading of data

chiamyunqing committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add command to enable printing of tasks occurring on a specific date
  Add feature to process date and time for events and deadlines

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

chiamyunqing committed:
Abstract out the parser, UI and TaskList classes

chiamyunqing committed:
Add JUnit tests for TaskList and Parser classes

chiamyunqing committed:
Add more tests for TaskListTest

chiamyunqing committed:
Include Javadocs comments to code

chiamyunqing committed:
Comply with coding standard

chiamyunqing committed:
Add feature to allow users to find tasks with keyword search

chiamyunqing committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Include Javadocs comments to code

chiamyunqing committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Comply with coding standard

# Conflicts:
#	src/main/java/Event.java
#	src/main/java/InvalidDateAndTimeException.java
#	src/main/java/Parser.java

chiamyunqing committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add feature to allow users to find tasks with keyword search

# Conflicts:
#	src/main/java/Duke.java
#	src/main/java/Parser.java
#	src/main/java/Ui.java

chiamyunqing committed:
Update test cases for text-ui-testing

chiamyunqing committed:
Merge commit '3b19ba1790d6035a9e246f5d4ae627bf68778606'

* commit '3b19ba1790d6035a9e246f5d4ae627bf68778606':
  Add Gradle support

chiamyunqing committed:
Successful set up of Gradle to run Duke and unit tests

chiamyunqing committed:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Successful set up of Gradle to run Duke and unit tests

chiamyunqing committed:
Use checkStyle with Gradle and edit accordingly but no Javadoc for constructors

chiamyunqing committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Use checkStyle with Gradle and edit accordingly but no Javadoc for constructors

chiamyunqing committed:
Update messages due to change in bot's personality to Doraemon

chiamyunqing committed:
Complete interaction with users via GUI

chiamyunqing committed:
Improve GUI by manipulating DialogBox

chiamyunqing committed:
Convert to using fxml

caitlinjee commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+     * Returns an Optional of the deadline date.
+     * @return an Optional of deadline date.
+     */
+    public Optional<LocalDate> getDate() {
```
Good idea to use optional, I didn't consider it for the tasks!

caitlinjee commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,185 @@
+     * Replace this stub with your completed method.
+     */
+    //ignore hi and bye first, this takes in one input and churn a response
+    private String getResponse(String input) {
```
This method seems a bit long, maybe you could separate out what to do in each case into different methods? For exmaple a handleList() method.

caitlinjee commented on src/main/java/MainWindow.java:
```diff
@@ -0,0 +1,108 @@
+
+ /*
+    @Override
+    public void start(Stage stage) {
```
Should this code be removed, since it's no longer in use?

caitlinjee commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,90 @@
+     * @throws NoDescriptionException If there is no task description for addition of tasks.
+     * @throws NotTaskException If an invalid command is being input.
+     */
+    public String[] parse(String command) throws InvalidDateAndTimeException,
```
Same with the previous switch statement, maybe you could separate out some of the code into separate methods to make this method shorter?

caitlinjee commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,24 @@
+
+    /**
+     * Returns an empty Optional since a to-do task does not have a date.
+     * @return an emptu Optional.
```
I think your empty has a typo >.<

caitlinjee reviewed (commented):
Overall looks good! Good idea using optional for the dates! Maybe the methods with the switch statements could be shorter. I like the bot's responses, they're really cute!

chiamyunqing committed:
Complete functionality for hi and bye messages

chiamyunqing committed:
Merge branch 'branch-A-Level-10'

* branch-A-Level-10:
  Complete functionality for hi and bye messages

chiamyunqing committed:
Edit the bug of exit button overlapping send button

thutahw commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,110 @@
+                + "I assume that you are as forgetful as Nobita. Remember to bribe me with loads of Dorayaki too!";
+        return intro;
+    }
+
```
Would it be better to name this method as "showWelcomeMessage"?

thutahw commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,110 @@
+        String bye = "Goodbye, I need to find my sister now! >_<";
+        return bye;
+    }
+
```
Would it be better if "showByeMessage" is used instead?

thutahw commented on src/test/java/TaskListTest.java:
```diff
@@ -0,0 +1,56 @@
+        listOfTask.add(new Todo("study"));
+        assertEquals(2, new TaskList(listOfTask).getNumTasks());
+    }
+
```
Maybe add a spacing here: "testGetNumTask() {" ? Also, would "listOfTasks" look better?

thutahw commented on src/test/java/TaskListTest.java:
```diff
@@ -0,0 +1,56 @@
+        assertEquals(listOfTask, new TaskList(listOfTask).getSameDateTasks("2020-08-08"));
+        assertEquals(new ArrayList<Task>(), new TaskList(listOfTask).getSameDateTasks("2020-08-09"));
+    }
+
```
I think it's the same as above, just missing 1 spacing between ")" and "{".

thutahw commented on src/test/java/TaskListTest.java:
```diff
@@ -0,0 +1,56 @@
+        assertEquals(listOfTask, new TaskList(new ArrayList<Task>(listOfTask)).getAllTasks());
+        assertEquals(new ArrayList<Task>(), new TaskList().getAllTasks());
+    }
+}
```
Everything is good! Maybe just the missing spacing like above.

thutahw reviewed (commented):
Overall it looks good! Just minor things like the missing spacing. Interesting use of Optional too!

chiamyunqing committed:
Add assertions to make sure objects are present at those states

Most issues arise from user mistakes and have already been handled by exceptions

chiamyunqing committed:
Improve code quality (e.g. comply with SLAP, prominent happy path)

chiamyunqing committed:
Merge pull request #1 from chiamyunqing/branch-A-Assertions

Add assertions to make sure objects are present at those states

chiamyunqing committed:
Merge branch 'master' into branch-A-CodeQuality

chiamyunqing committed:
Merge pull request #2 from chiamyunqing/branch-A-CodeQuality

Improve code quality (e.g. comply with SLAP, prominent happy path)

chiamyunqing committed:
Use streams in TaskList

chiamyunqing committed:
Merge pull request #3 from chiamyunqing/branch-A-Streams

Use streams in TaskList

chiamyunqing committed:
Create gradle.yml

chiamyunqing committed:
Refactor code in Parser using extract method

chiamyunqing committed:
Add feature for note adding to tasks

chiamyunqing committed:
Comply with checkstyle

chiamyunqing committed:
Update test code to be aligned with updated Task class

chiamyunqing committed:
Improve GUI for exit button

chiamyunqing committed:
Comply with checkstyle

chiamyunqing committed:
Add comments to give credits for solutions adapted from others

chiamyunqing committed:
Add Ui.png

chiamyunqing committed:
Update README.md half way

chiamyunqing committed:
Update README.md

chiamyunqing committed:
Update README.md

chiamyunqing committed:
Update README.md

chiamyunqing committed:
Update README.md

chiamyunqing committed:
Update README.md

chiamyunqing committed:
Update task description to make sure find works very well

chiamyunqing committed:
Set theme jekyll-theme-leap-day

chiamyunqing committed:
Update README.md

chiamyunqing committed:
Update README.md
